irispy/spectrograph.py

```diff
@@ -54,18 +54,34 @@ def __init__(self, data, meta=None):
     def __repr__(self):
         spectral_window = self.spectral_windows["spectral window"][0]
         spectral_windows_info = "".join(
-            ["\n    {0}\n        (raster axis, slit axis, spectral axis) {1}".format(
-                name,
-                self.data[name].dimensions[1::])
+            ["\n{0:>15} : {1:16} pix".format(name,
+                str([int(dim.value) for dim in self.data[name].dimensions[1::]]))
                 for name in self.spectral_windows["spectral window"]])
-        return "<iris.IRISSpectrograph instance\nOBS ID: {0}\n".format(self.meta["OBSID"]) + \
-               "OBS Description: {0}\n".format(self.meta["OBS_DESC"]) + \
-               "OBS period: {0} -- {1}\n".format(self.meta["STARTOBS"], self.meta["ENDOBS"]) + \
-               "Instance period: {0} -- {1}\n".format(
-                   self.data[spectral_window][0].extra_coords["time"]["value"],
-                   self.data[spectral_window][-1].extra_coords["time"]["value"]) + \
-               "Number unique raster positions: {0}\n".format(self.meta["NRASTERP"]) + \
-               "Spectral windows{0}>".format(spectral_windows_info)
+        obs_start = self.meta["STARTOBS"]
+        obs_end = self.meta["ENDOBS"]
+        time_start = self.data[spectral_window][0].extra_coords["time"]["value"].min()
+        time_end = self.data[spectral_window][-1].extra_coords["time"]["value"].max()
+        result = []
+        # If starting and ending times in same day, print date only once
+        for start, end in zip([obs_start, time_start], [obs_end, time_end]):
+            if start.strftime("%j") == end.strftime("%j"):
+                result.append("{0} {1} -- {2}".format(
+                    start.strftime("%Y-%m-%d"), start.strftime("%H:%M:%S.%f"),
+                    end.strftime("%H:%M:%S.%f")))
+            else:
+                result.append("{0} -- {1}".format(start, end))
+        obs_period, instance_period = result
+        return ("<iris.IRISSpectrograph instance\nOBS ID: {obsid}\n"
+               "OBS Description: {obsdesc}\n"
+               "OBS period: {obs_period}\n"
+               "Instance period: {inst_period}\n"
+               "Number unique raster positions: {nraster}\n"
+               "Number of repeats: {repeats}\n"
```

Comment:
I think there's a case for going with something like ```OBS Number unique raster positions``` and ```OBS Number of repeats``` to make it clear these metadata refer to the the OBS as a whole, rather than the data in the ```IRISSpectrograph```.  If the data are sliced, or if only some of the files from the OBS were read in, these numbers will disagree with the shape of the data as the are taken from metadata and could be confusing.